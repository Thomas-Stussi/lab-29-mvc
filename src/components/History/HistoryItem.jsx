import React from 'react';
import PropTypes from 'prop-types';

const HistoryItem = ({ url, method, body }) => (
  <>
    <h3>{method}</h3>
    <p>{url}</p>
    <p>{body}</p>
  </>
);

HistoryItem.propTypes = {
  url: PropTypes.string.isRequired,
  body: PropTypes.string,
  method: PropTypes.string.isRequired
};

export default HistoryItem;
