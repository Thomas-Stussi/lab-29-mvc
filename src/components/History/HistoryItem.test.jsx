import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryItem from './HistoryItem';

describe('HistoryItem component', () => {
  afterEach(() => cleanup());
  it('render HistoryItem component', () => {
    const { asFragment } = render(<HistoryItem
      url="https://www.thecocktaildb.com/api/json/v1/1/random.php"
      method="GET"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
