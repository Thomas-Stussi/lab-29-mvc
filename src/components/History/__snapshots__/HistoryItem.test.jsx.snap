// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryItem component render HistoryItem component 1`] = `
<DocumentFragment>
  <h3>
    GET
  </h3>
  <p>
    https://www.thecocktaildb.com/api/json/v1/1/random.php
  </p>
  <p />
</DocumentFragment>
`;
